app/controllers/candidate_interface/gcse_details_controller.rb

```diff
@@ -1,15 +1,83 @@
 module CandidateInterface
   class GcseDetailsController < CandidateInterfaceController
-    def edit
-      subject = params[:subject]
+    # 1th step - Edit qualification type
+    def edit_type
+      @subject = subject_param
 
-      @application_qualification = ApplicationQualification.create(subject: subject, application_form: current_candidate.current_application, level: 'gcse')
+      @application_qualification = GcseQualificationTypeForm.new(
+        qualification_type: '',
+        subject: subject_param,
+        level: 'gcse',
+      )
     end
 
-    def update
+    def update_type
+      @subject = subject_param
+      @application_qualification = GcseQualificationTypeForm
+                                     .new(qualification_type: (params[:candidate_interface_gcse_qualification_type_form] || {}).fetch(:qualification_type, ''),
+                                          subject: subject_param,
+                                          level: 'gcse')
```

Comment:
This line is quite difficult to read, could we refactor this and potentially `params[:candidate_interface_gcse_qualification_type_form]` into a private method?